### DISCUSSION TITLE
Change location of alire/ directory

### DISCUSSION THREAD

**Discussion by dinkelk** [alire-project/alire#1549] [Ideas]
By default `alr` stores dependencies and other configuration in an `alire/` directory, created at the top of a user's repository. Is there a way to change the name and location of this directory, maybe via command line argument? If not, is this a feature you would consider adding?

**Use case:** I often develop in docker containers with my source repository bind mounted to the container, so code can be edited on the host, but compiled in the container. On many host machines, the more files that exist in this bind mount, the slower compilation and other operations on the bind mount become. If the `alire/` directory could live somewhere else on the container, and not in the bind mount, then all those dependency files would not slow down the container.

**Comment by mosteo** [comment]
Dependency files won't be stored under `alire/` after the 2.0 release, only a few files will remain by default, so that should alleviate the problem.

If you use pins that's another matter, these will continue living under `alire/`.

If you move the `alire/` folder and softlink it, is it still available inside the container?

**Comment by dinkelk** [comment]
This is great info about the 2.0 release. And your symlink suggestion does indeed work. Thank you!

**Comment by mgrojo** [comment]
> **Use case:** I often develop in docker containers with my source repository bind mounted to the container, so code can be edited on the host, but compiled in the container. On many host machines, the more files that exist in this bind mount, the slower compilation and other operations on the bind mount become. If the `alire/` directory could live somewhere else on the container, and not in the bind mount, then all those dependency files would not slow down the container.

To alleviate this in any case, you can use the [`.dockerignore`](https://docs.docker.com/build/building/context/#dockerignore-files) file.
